tethysext/atcore/models/resource_workflow_results/plot_workflow_result.py

```diff
@@ -0,0 +1,127 @@
+"""
+********************************************************************************
+* Name: dataset_workflow_result.py
+* Author: nswain
+* Created On: June 3, 2019
+* Copyright: (c) Aquaveo 2019
+********************************************************************************
+"""
+import copy
+import pandas as pd
+from tethysext.atcore.models.app_users.resource_workflow_result import ResourceWorkflowResult
+import plotly.graph_objs as go
+
+__all__ = ['PlotWorkflowResult']
+
+
+class PlotWorkflowResult(ResourceWorkflowResult):
+    """
+    Data model for storing spatial information about resource workflow results.
+    """
+    CONTROLLER = 'tethysext.atcore.controllers.resource_workflows.results_views.plot_workflow_results_view.PlotWorkflowResultView'  # noqa: E501
+    TYPE = 'plot_workflow_result'
+
+    __mapper_args__ = {
+        'polymorphic_on': 'type',
+        'polymorphic_identity': TYPE
+    }
+
+    def __init__(self, *args, **kwargs):
+        """
+        Constructor.
+
+        Args:
+        """
+        super().__init__(*args, **kwargs)
+
+    @property
+    def default_options(self):
+        """
+        plot_lib (str): bokeh or plotly
+        axes(list): A list of tuples for pair axis ex. For example: [('x', 'y'), ('x1', 'y1'), ('x', 'y2')]
+        labels(list): A list of series' label. For example: ['Series 1', 'Series 2', 'Series 3']
+        axis_labels(list): A list of label for x and y axes respectively. For example: ['x', 'y']
+        plot_type (str): lines or scatter
+        line_shape (str): Only for plotly. You can select from on of these options: linear, spline, vhv, hvh, vh, hv
+
+        Returns default options dictionary for the object.
+        """
+        default_options = super().default_options
+
+        default_options.update({
+            'plot_lib': 'plotly',
+            'axes': [],
+            'labels': [],
+            'plot_type': 'lines',
+            'axis_labels': ['x', 'y'],
+            'line_shape': 'linear',
+            'no_dataset_message': 'No dataset found.'
+        })
+        return default_options
+
+    @property
+    def datasets(self):
```

Comment:
Should we support adding multiple datasets? if not, then this property should probably be called 'dataset' (singular). See comment on `add_pandas_dataframe` method.

Reply:
Looking at the logic in the view, it looks like the `plot_view` object is overwritten in each loop, which means the plot will show only the last dataframe. Refactor this so that we can only assign one dataset, not add multiple.

Reply:
Alternatively, we could allow multiple dataframes and use them to define series instead of using the `axes` and `labels` variables. The title of the dataset would be the name of the series.